src/Polyfill/Lock.cs

```diff
@@ -0,0 +1,182 @@
+// <auto-generated />
+#pragma warning disable
+
+#nullable enable
+
+#if !NET9_0_OR_GREATER
+
+namespace System.Threading;
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Link = System.ComponentModel.DescriptionAttribute;
+
+/// <summary>
+/// Provides a way to get mutual exclusion in regions of code between different threads. A lock may be held by one thread at
+/// a time.
+/// </summary>
+/// <remarks>
+/// Threads that cannot immediately enter the lock may wait for the lock to be exited or until a specified timeout. A thread
+/// that holds a lock may enter the lock repeatedly without exiting it, such as recursively, in which case the thread should
+/// eventually exit the lock the same number of times to fully exit the lock and allow other threads to enter the lock.
+/// </remarks>
+[ExcludeFromCodeCoverage]
+[DebuggerNonUserCode]
+[Link("https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock")]
+#if PolyPublic
+public
+#endif
+class Lock
+{
+#if (NETCOREAPP) || (NETFRAMEWORK && NET45_OR_GREATER) || (NETSTANDARD)
```

Comment:
Out of curiosity, what _isn't_ covered by this `#if`?

Reply:
Earliest supported is `net461`
https://github.com/SimonCropp/Polyfill/blob/34a475e1a069b89811801ef0ed1d5b8580c9b449/readme.md?plain=1#L8-L12

Reply:
That's a bit ambiguous to be honest because the earliest .NET Framework targeted by .NET Standard 2.0 is .NET Framework 4.5 (which has `IsHeldByCurrentThread`) and not .NET Framework 4.6.1.

Reply:
Sorry, I misread.

Yes, in that case the use of `IsHeldByCurrentThread` is available. It is available since .NET Standard 1.0 (.NET Framework 4.5)